src/charm.py

```diff
@@ -32,24 +32,12 @@ def __init__(self, *args):
         self.framework.observe(self.on.config_changed, self._on_config_changed)
         self.framework.observe(self.on.gosherve_pebble_ready, self._on_config_changed)
         self.framework.observe(self.on.pull_site_action, self._pull_site_action)
-
-        self.ingress = IngressRequires(
+        self.ingress = IngressPerAppRequirer(
             self,
-            {
-                "service-hostname": self._external_hostname,
-                "service-name": self.app.name,
-                "service-port": 8080,
-            },
+            host=f"{self.app.name}-endpoints.{self.model.name}.svc.cluster.local",
```

Comment:
```suggestion
            host=f"{self.app.name}.{self.model.name}.svc.cluster.local",
```

I don't think we should use the `endpoints` svc here

Reply:
In order to use the `service/hello-kubecon`, charm need to use the open-port to adjust port specs on the service.

open-port is only available on juju 3.1.0 above, if we use the `service/hello-kubecon` instead of `service/hello-kubecon-endpoints`, this charm will not be able to run on juju < 3.1.0. Is this acceptable?

Reply:
I don't think that's true? I think the easiest route here is just to do:

```python
self.ingress = IngressPerAppRequirer(self, port=8080)
```

Reply:
If `host` is not provided in the `IngressPerAppRequirer`, the `host` will be retrieved by `socket.getfqdn()`. In terms of the hello-kubecon charm, it will be `hello-kubecon-0.hello-kubecon-endpoints.test-hello.svc.cluster.local`

Reply:
Okay, I'll pick that up seperately. Thanks.